wandb/__init__.pyi

```diff
@@ -416,48 +416,93 @@ def log(
 
-    In the UI, summary values show up in the run table to compare single values across runs.
-    Summary values can also be set directly with `wandb.run.summary["key"] = value`.
-
     Logged values don't have to be scalars. Logging any wandb object is supported.
-    For example `wandb.log({"example": wandb.Image("myimage.jpg")})` will log an
+    For example `run.log({"example": wandb.Image("myimage.jpg")})` will log an
     example image which will be displayed nicely in the W&B UI.
     See the [reference documentation](https://docs.wandb.com/ref/python/data-types)
     for all of the different supported types or check out our
     [guides to logging](https://docs.wandb.ai/guides/track/log) for examples,
     from 3D molecular structures and segmentation masks to PR curves and histograms.
-    `wandb.Table`s can be used to logged structured data. See our
+    You can use `wandb.Table` to log structured data. See our
     [guide to logging tables](https://docs.wandb.ai/guides/data-vis/log-tables)
     for details.
 
     Logging nested metrics is encouraged and is supported in the W&B UI.
-    If you log with a nested dictionary like `wandb.log({"train":
-    {"acc": 0.9}, "val": {"acc": 0.8}})`, the metrics will be organized into
+    When you log with a nested dictionary like `run.log({"train":
+    {"acc": 0.9}, "val": {"acc": 0.8}})`, the metrics are organized into
     `train` and `val` sections in the W&B UI.
 
-    wandb keeps track of a global step, which by default increments with each
-    call to `wandb.log`, so logging related metrics together is encouraged.
-    If it's inconvenient to log related metrics together
-    calling `wandb.log({"train-loss": 0.5}, commit=False)` and then
-    `wandb.log({"accuracy": 0.9})` is equivalent to calling
-    `wandb.log({"train-loss": 0.5, "accuracy": 0.9})`.
+    `run.log` is not intended to be called more than a few times per second.
+
+    ### Summary values
+
+    The UI shows summary values in the run overview page
+    and in the run table for comparison.
+    Summary values can be set directly by using the `summary` property:
+
+    ```
+    run.log({"train": {"loss": 100}})
+    run.log({"train": {"loss": 90}})
+    run.log({"train": {"loss": 98}})
+    run.summary["train"]["loss"] = 90
+    ```
+
+    See also `define_metric` for more control over summary calculations.
+
+    ### The W&B step
+
+    With basic usage, each call to `log` creates a new "step".
+    The step must always increase, and it is not possible to log
+    to a previous step.
 
-    `wandb.log` is not intended to be called more than a few times per second.
-    If you want to log more frequently than that it's better to aggregate
-    the data on the client side or you may get degraded performance.
+    Note that you can use any metric as the X axis in charts.
+    In many cases, it is better to think of the W&B step as
```

Comment:
can you help me understand this piece a bit better? more so I can build some context on it

Reply:
A lot of user confusion is caused by the name "step" sounding like it should be related to a training step or iteration, but it's really more like an index into a database. It's an unfortunate historical mistake that's hard to change now, but that I'd love to improve it for 1.0.

Some users try to set the step manually and get frustrated by the various restrictions on it. Those users should probably be logging a separate metric as an X axis instead of setting the step, but there are other problems with this (in particular run forking is unfortunately designed around the step AFAIK).

Reply:
very accurate! probably most of the forking would be redundant if people could just plot against any step and view all historical values against said step...